framework/src/meshgenerators/StackGenerator.C

```diff
@@ -0,0 +1,161 @@
+  _meshes.reserve(_input_names.size() - 1);
+
+  // Read in all of the other meshes
+  for (auto i = beginIndex(_input_names, 1); i < _input_names.size(); ++i)
+    _meshes.push_back(dynamic_pointer_cast<ReplicatedMesh>(*_mesh_ptrs[i]));
+
+  // Check that the casts didn't fail, and that the dimensions match
+  for (auto i = beginIndex(_meshes); i < _meshes.size(); ++i)
+  {
+    if (_meshes[i] == nullptr)
+      mooseError("StackGenerator only works with ReplicatedMesh : mesh from Meshgenerator ",
+                 _input_names[i + 1],
+                 "is not a ReplicatedMesh.");
+    if (int(_meshes[i]->mesh_dimension()) != dim)
+      mooseError("Mesh from MeshGenerator : ", _input_names[i + 1], " is not in ", _dim, "D.");
+  }
+
+  boundary_id_type first =
+      mesh->get_boundary_info().get_id_by_name(getParam<BoundaryName>("top_boundary"));
+  boundary_id_type second =
+      mesh->get_boundary_info().get_id_by_name(getParam<BoundaryName>("bottom_boundary"));
+
+  if (dim == 3)
+  {
+    first = mesh->get_boundary_info().get_id_by_name(getParam<BoundaryName>("front_boundary"));
+    second = mesh->get_boundary_info().get_id_by_name(getParam<BoundaryName>("back_boundary"));
+  }
+
+  // Getting the width of each mesh
+  std::vector<Real> heights;
+  heights.push_back(computeWidth(*mesh, _dim) + _bottom_height);
+  for (auto i = beginIndex(_meshes); i < _meshes.size(); ++i)
+    heights.push_back(computeWidth(*_meshes[i], _dim) + *heights.rbegin());
+
+  // Move the first mesh at the provided height
+  switch (_dim)
+  {
+    case 2:
+      MeshTools::Modification::translate(*mesh, 0, _bottom_height, 0);
+      break;
+    case 3:
+      MeshTools::Modification::translate(*mesh, 0, 0, _bottom_height);
+      break;
+  }
+
+  // Move all of the other meshes in the right spots then stitch them one by one to the first one
+  for (auto i = beginIndex(_meshes); i < _meshes.size(); ++i)
+  {
+    switch (_dim)
+    {
+      case 2:
+        MeshTools::Modification::translate(*_meshes[i], 0, heights[i], 0);
+        break;
+      case 3:
+        MeshTools::Modification::translate(*_meshes[i], 0, 0, heights[i]);
+        break;
+    }
+    mesh->stitch_meshes(
+        *_meshes[i], first, second, TOLERANCE, /*clear_stitched_boundary_ids=*/true);
+  }
+
+  return dynamic_pointer_cast<MeshBase>(mesh);
+}
+
+Real
+StackGenerator::computeWidth(const MeshBase & mesh, const int & dim)
+{
+  std::set<subdomain_id_type> sub_ids;
+  mesh.subdomain_ids(sub_ids);
+  BoundingBox bbox(Point(std::numeric_limits<Real>::max(),
+                         std::numeric_limits<Real>::max(),
+                         std::numeric_limits<Real>::max()),
+                   Point(std::numeric_limits<Real>::lowest(),
+                         std::numeric_limits<Real>::lowest(),
+                         std::numeric_limits<Real>::lowest()));
+  for (auto id : sub_ids)
+  {
+    BoundingBox sub_bbox = MeshTools::create_subdomain_bounding_box(mesh, id);
```

Comment:
I don't understand.  Can you explain all the stuff with the subdomains?  It doesn't make sense to me.  Why can't you just call `MeshTools::create_bounding_box()`?  It should give you the exactly correct bounding box (no need to union or anything).

But what am I missing?